### PR TITLE
fixing warnings in tests code, part2

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: test/test_resume.cpp [hunk 1]
```diff
@@ -77,7 +79,7 @@ std::shared_ptr<torrent_info> generate_torrent()
 	for (piece_index_t i(0); i < fs.end_piece(); ++i)
 	{
 		sha1_hash ph;
-		for (int k = 0; k < 20; ++k) ph[k] = lt::random(0xff);
+		for (std::size_t k = 0; k < 20; ++k) ph[k] = lt::random(0xff) & 0xff;
```
Comment (arvidn) [inline test/test_resume.cpp]: what do you think of: ``for (auto& v : ph) v = lt::random(0xff) & 0xff);``?

Comment (aldenml) [reply test/test_resume.cpp]: or just `aux::random_bytes(ph);` :)

**Comment by arvidn** [comment]
I cancelled the last osx travis job to get one of my other two PRs to run. I trust this one is fine, but I don't want to merge it until the other two PRs have landed (which cuts down on OSX builds).

**Comment by aldenml** [comment]
sure, no problem

**Pull request merged**